src/core/gltf2importer/gltf2options.cpp

```diff
@@ -0,0 +1,76 @@
+/*
+    gltf2options.cpp
+
+    This file is part of Kuesa.
+
+    Copyright (C) 2018-2019 Klarälvdalens Datakonsult AB, a KDAB Group company, info@kdab.com
+    Author: Juan Casafranca <juan.casafranca@kdab.com>
+
+    Licensees holding valid proprietary KDAB Kuesa licenses may use this file in
+    accordance with the Kuesa Enterprise License Agreement provided with the Software in the
+    LICENSE.KUESA.ENTERPRISE file.
+
+    Contact info@kdab.com if any conditions of this licensing are not clear to you.
+
+    This program is free software: you can redistribute it and/or modify
+    it under the terms of the GNU Affero General Public License as
+    published by the Free Software Foundation, either version 3 of the
+    License, or (at your option) any later version.
+
+    This program is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU Affero General Public License for more details.
+
+    You should have received a copy of the GNU Affero General Public License
+    along with this program.  If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#include "gltf2options.h"
+
+/*!
+ * \class Kuesa::GLTF2Import::GLTF2Options
+ * \inheaderfile Kuesa/GLTF2Options
+ * \inmodule Kuesa
+ * \since 1.1
+ * \brief Different options to configure the glTF 2.0 importer
+ *
+ * Avalaible options are
+ * \list
+ * \li generateTangents: If true, the importer will generate tangents
+ * for all the primitives which doesn't have tangents.
+ * \endlist
+ */
+
+/*!
+ * \qmltype GLTF2Options
+ * \instantiates Kuesa::GLTF2Import::GLTF2Options
+ * \inmodule Kuesa
+ * \since 1.1
+ * \brief Different options to configure the glTF 2.0 importer
+ *
+ * Avalaible options are
+ * \list
+ * \li generateTangents: If true, the importer will generate tangents
+ * for all the primitives which doesn't have tangents.
+ * \endlist
+ */
+
+Kuesa::GLTF2Import::GLTF2Options::GLTF2Options()
```

Comment:
Should the ctor allow to receive a parent in so as to have its lifetime properly managed ?

Reply:
In fact, it feels really wrong the I need to dynamically allocate and manage the lifetime of this object. It should be just a plain structure with the options, but as I need to be able to create it from qml and to emit signals, I need to derive from QObject and make it dynamic. Isnt't there a better approach?

Reply:
Refactor so that we dont need to allocate a GLTF2Options object.